lib/core-integration/src/Test/Integration/Scenario/API/Shelley/StakePools.hs

```diff
@@ -679,16 +679,21 @@ spec = do
                             , description = Just "Lorem Ipsum Dolor Sit Amet."
                             , homepage = "https://iohk.io"
                             }
+                        , StakePoolMetadata
+                            { ticker = (StakePoolTicker "GPD")
+                            , name = "Genesis Pool D"
+                            , description = Just "Lorem Ipsum Dolor Sit Amet."
+                            , homepage = "https://iohk.io"
+                            }
                         ]
 
                 verify r
                     [ expectListSize 3
-                    , expectField Prelude.id $ \pools ->
-                        -- To ignore the arbitrary order,
-                        -- we sort on the names before comparing
-                        sortOn name
-                            (mapMaybe (fmap getApiT . view #metadata) pools)
-                            `shouldBe` metadataList
+                    , expectField Prelude.id $ \pools -> do
+                        let metadataActual = Set.fromList $
+                                mapMaybe (fmap getApiT . view #metadata) pools
+                        metadataActual
+                            `shouldSatisfy` (`Set.isSubsetOf` metadataPossible)
```

Comment:
Nitpicking (:sweat_smile:), but maybe we could add a stronger assertion that the list of returned pools does not contain GPD?

Reply:
> Nitpicking (sweat_smile), but maybe we could add a stronger assertion that the list of returned pools does not contain GPD?

Unfortunately we can't currently do this, as there is no guarantee that pool `GPD` will be the one to be assigned with an early retirement epoch. It could be _any_ of the four pools that gets retired early.

Reply:
I suspect we could solve this problem entirely by changing our test framework to allow pools to be registered and retired dynamically, as in: https://jira.iohk.io/browse/ADP-385